### PR TITLE
Use datadog for monitoring

### PR DESCRIPTION
**Pull request by tom-blake**
Use Datadog for monitoring to replace Hotjar. Use Vite's mode to switch environment configuration. Add environment variable to toggle Datadog integration for developers.

Test plan:
- Confirm local logs are not shown in datadog by default.
- Confirm dev and prod logs are sent to datadog
- Confirm env and version tags on logs are correctly set.

### CODE REVIEW

File: src/layouts/BaseLayout.astro [hunk 1]
```diff
@@ -49,12 +53,12 @@ const navigation = Navigation.create({ nav: sideNavConfig, content });
     <meta name="viewport" content="width=device-width">
     <title>{metaTitle}</title>
     <meta name="description" content={description}>
-    <meta property="img" content={`${Astro.url.origin}${img}`}>
+    <meta property="img" content={img}>
```
Comment (MeganSteenkamp) [inline src/layouts/BaseLayout.astro:56]: Question: Was this change intended?

**Review by MeganSteenkamp — APPROVED** [review]

File: src/layouts/BaseLayout.astro (hunk 1 above)
Comment (tom-blake) [reply src/layouts/BaseLayout.astro:56]: @MeganSteenkamp Yep - converting these to relative links avoids complexity when switching between environments
